src/protocol/payload/block.rs

```diff
@@ -46,12 +47,20 @@ impl LocatorHashes {
     }
 }
 
+#[derive(Debug)]
 pub struct Headers {
     count: VarInt,
     headers: Vec<Header>,
 }
 
 impl Headers {
+    pub fn empty() -> Self {
```

Comment:
I like how explicit this is, we may want to add `Default` implementations in future if this is a pattern we find ourselves using a lot. Just something to keep in mind.